File: src/types/habit.ts
export type HabitTrackingType = 'boolean' | 'counter' | 'duration' | 'quantity' | 'rating'
export type HabitDirection = 'build' | 'limit'

export interface Habit {
  id: string
  name: string
  description: string | null
  icon: string | null
  color: string
  tracking_type: HabitTrackingType
  unit: string | null
  daily_goal: number | null
  direction: HabitDirection
  is_archived: number  // 0 | 1
  sort_order: number
  created_at: string
  updated_at: string
  deleted_at: string | null
  sync_status: string
}

export interface HabitEntry {
  id: string
  habit_id: string
  entry_date: string  // YYYY-MM-DD
  value: number
  note: string | null
  created_at: string
  updated_at: string
  deleted_at: string | null
  sync_status: string
}

export interface CreateHabitPayload {
  name: string
  description?: string | null
  icon?: string | null
  color: string
  tracking_type: HabitTrackingType
  unit?: string | null
  daily_goal?: number | null
  direction: HabitDirection
}

export interface UpdateHabitPayload {
  id: string
  name?: string | null
  description?: string | null
  icon?: string | null
  color?: string | null
  unit?: string | null
  daily_goal?: number | null
  direction?: string | null
  is_archived?: boolean | null
  sort_order?: number | null
}

export interface UpsertHabitEntryPayload {
  habit_id: string
  entry_date: string
  value: number
  note?: string | null
}

File: src-tauri/src/models/habit.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct Habit {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    pub icon: Option<String>,
    pub color: String,
    pub tracking_type: String,
    pub unit: Option<String>,
    pub daily_goal: Option<f64>,
    pub direction: String,
    pub is_archived: i64,
    pub sort_order: i64,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: Option<String>,
    pub sync_status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct HabitEntry {
    pub id: String,
    pub habit_id: String,
    pub entry_date: String,
    pub value: f64,
    pub note: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    pub deleted_at: Option<String>,
    pub sync_status: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateHabitPayload {
    pub name: String,
    pub description: Option<String>,
    pub icon: Option<String>,
    pub color: String,
    pub tracking_type: String,
    pub unit: Option<String>,
    pub daily_goal: Option<f64>,
    pub direction: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateHabitPayload {
    pub id: String,
    pub name: Option<String>,
    pub description: Option<String>,
    pub icon: Option<String>,
    pub color: Option<String>,
    pub unit: Option<String>,
    pub daily_goal: Option<f64>,
    pub direction: Option<String>,
    pub is_archived: Option<bool>,
    pub sort_order: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpsertHabitEntryPayload {
    pub habit_id: String,
    pub entry_date: String,
    pub value: f64,
    pub note: Option<String>,
}
